src/dctest/expressions.cljs

```diff
@@ -284,7 +287,16 @@ ExpectedInterpolation ::= InterpolatedExpression PrintableChar*
 
                      ;; Recurse
                      (recur (zip/next loc)
-                            refs)))))]
-    (reduce #(assoc %1 %2 (read-eval context %2))
+                            refs)))))
+        safe-read-eval (fn [context expr]
+                         (let [{:keys [result error-msg]}
+                               , (try
+                                   {:result (read-eval context expr)}
+                                   (catch js/Error err
+                                     {:error-msg (.-message err)}))]
+                           (if error-msg
+                             (str "<Error: " error-msg ">")
+                             result)))]
+    (reduce #(assoc %1 %2 (safe-read-eval context %2))
```

Comment:
Are `%1` and `%2` always maps that can be `assoc`d?  In the case of an error, isn't it just a string?

Reply:
`%1` is a map (expressions to values/results) that we're building up, and `%2` is the expression we are evaluating (which could result in a string, exactly).